app/javascript/react/components/ReviewTile.js

```diff
@@ -80,10 +80,49 @@ const ReviewTile = props =>{
     }
   }
 
+  const deleteReviewFetch = async () => {
+    try {
+      const response = await fetch(`/api/v1/sauces/${props.sauce.id}/reviews/${props.id}`,{
+        credentials: "same-origin",
+        method: "DELETE",
+        headers: {
+          "Accept": "application/json",
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify({user_id: props.reviewUserId})
+      }) 
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.status.text})`
+        const error = new Error(errorMessage)
+        throw(error)
+      }
+      const sauceObject = await response.json()
+      props.setSauce(sauceObject.sauce)
+    } catch (error) {
+      console.log(`Error in fetch: ${error}`)
+    }
+  }
+
+  const deleteReviewClickHandler = (event) => {
+    event.preventDefault()
+    deleteReviewFetch()
+  }
+
+  const reviewDeleteButton = () => {
+    if (props.currentUser.role === 'admin' || props.currentUser.id === props.reviewUserId) {
+      return (
+        <button className="button" onClick={deleteReviewClickHandler}>Delete this review</button>
+      )
+    } 
+  }
```

Comment:
this doesnt need to be a function. Just have the if statement on the outside, but assign the button element to either be null or equal to what you have on line 114

Reply:
removed function wrapper and just return null if the user is not allowed to delete the review